runtime/onert/core/src/compiler/StaticShapeInferer.cc

```diff
@@ -605,6 +605,23 @@ void StaticShapeInferer::visit(const ir::operation::L2Normalization &op)
   handleSimpleUnaryOp(op, op.getInputs().at(ir::operation::L2Normalization::Input::INPUT));
 }
 
+void StaticShapeInferer::visit(const ir::operation::Loss &op)
+{
+  auto &operands = _lowered_subg->graph().operands();
+
+  const auto y_pred_idx{op.getInputs().at(ir::operation::Loss::Input::Y_PRED)};
+  const auto &y_pred = operands.at(y_pred_idx);
+
+  const auto y_true_idx{op.getInputs().at(ir::operation::Loss::Input::Y_TRUE)};
+  auto &y_true = operands.at(y_true_idx);
+
+  // TODO Consider SparseCategoricalCrossentropy case
+
+  y_true.info().shape(y_pred.info().shape());
+
+  // TODO Consider output shape in case of reduction option
```

Comment:
Why don't you set the output shape in case it's not a reduction option?

Reply:
We don't need to change the loss output even if the batch size changes.
The output of loss is always a scalar tensor, regardless of batch_size.
However, some types of reduction option use different shape output size.
I will consider it later.